pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,222 @@
+		windows.Handle(0))
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Info("Connected to driver and handle created")
+	return h, nil
+}
+
+// CloseDriverHandle closes an open handle on the driver
+func (dc *DriverInterface) CloseDriverHandle(handle windows.Handle) error {
+	return windows.CloseHandle(handle)
+}
+
+// Add buffers to Tracer object. Even though the windows API will actually
+// keep the buffers, add it to the tracer so that golang doesn't garbage collect
+// the buffers out from under us
+func (dc *DriverInterface) prepareReadBuffers(bufs []readBuffer) ([]readBuffer, error) {
+	for i := 0; i < totalReadBuffers; i++ {
+		err := windows.ReadFile(dc.driverHandle, bufs[i].data[:], nil, &bufs[i].ol)
+		if err != nil {
+			if err != windows.ERROR_IO_PENDING {
+				fmt.Printf("failed to initiate readfile %v\n", err)
+				windows.CloseHandle(dc.iocp)
+				dc.iocp = windows.Handle(0)
+				return nil, err
+			}
+		}
+	}
+	return bufs, nil
+}
+
+func (dc *DriverInterface) prepareDriverFilters() (err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		log.Errorf("Error getting interfaces: %s\n", err.Error())
+		return
+	}
+
+	for _, i := range ifaces {
+		log.Debugf("Setting filters for interface: %s [%+v]", i.Name, i)
+
+		for _, filter := range createFiltersForInterface(i) {
+			err = dc.setFilter(filter)
+			if err != nil {
+				return log.Warnf("Failed to set filter [%+v] on interface [%+v]: %s\n", filter, i, err.Error())
+			}
+		}
+	}
+	return nil
+}
+
+// To capture all traffic for an interface, we create an inbound/outbound traffic filter
+// for both IPV4 and IPV6 traffic going to that interface
+func createFiltersForInterface(iface net.Interface) (filters []C.struct__filterDefinition) {
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, iface.Index, false))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, iface.Index, false))
+	return
+}
+
+// NewDDAPIFilter returns a filter we can apply to the driver
+func newDDAPIFilter(direction C.uint64_t, ifaceIndex int, isIPV4 bool) (fd C.struct__filterDefinition) {
+	fd.filterVersion = C.DD_FILTER_SIGNATURE
+	fd.size = C.sizeof_struct__filterDefinition
+	fd.direction = direction
+
+	if isIPV4 {
+		fd.af = windows.AF_INET
+		fd.v4InterfaceIndex = (C.ulonglong)(ifaceIndex)
+	} else {
+		fd.af = windows.AF_INET6
+		fd.v6InterfaceIndex = (C.ulonglong)(ifaceIndex)
+	}
+
+	return fd
+}
+
+func (dc *DriverInterface) setFilter(fd C.struct__filterDefinition) (err error) {
+	var id int64
+	err = windows.DeviceIoControl(dc.driverHandle, C.DDFILTER_IOCTL_SET_FILTER, (*byte)(unsafe.Pointer(&fd)), uint32(unsafe.Sizeof(fd)), (*byte)(unsafe.Pointer(&id)), uint32(unsafe.Sizeof(id)), nil, nil)
+	if err != nil {
+		return log.Errorf("Failed to set filter: %s\n", err.Error())
```

Comment:
`prepareDriverFilters` is already logging any error returned by setFilter, so a better approach to error handling would be to wrap the error with `fmt.Errorf("failed to set filter: %v", err)`. Same comment for the rest of this file. We should prefer logging one good error to logging a bunch of errors that are missing context.